run_api.py

```diff
@@ -0,0 +1,105 @@
+    temp_frame_quality: Optional[int] = 0
+    output_video_encoder: Optional[Literal['libx264', 'libx265', 'libvpx-vp9', 'h264_nvenc', 'hevc_nvenc']] = 'libx264'
+    output_video_quality: Optional[int] = 0
+    max_memory: Optional[int] = 0
+    execution_threads: Optional[int] = suggest_execution_threads()
+@app.get('/get_execution_providers')
+async def get_execution_provider():
+    return suggest_execution_providers()
+
+
+@app.post("/start_roop")
+async def image_file(
+        src_file: UploadFile = File(...),
+        target_file: UploadFile = File(...),
+        roop_parameters: RoopModel = Depends()
+):
+    # Removing the folder and its content if it already exists
+    saving_path = resolve_relative_path('../workdir/')
+    if os.path.exists(saving_path):
+        shutil.rmtree(saving_path)
+    os.makedirs(saving_path)
+
+    #Get execution provider from env.
+    execution_provider = os.getenv('EXECUTION_PROVIDER')
+    print("execution provider is set to {}".format(execution_provider))
+    if execution_provider == 'CUDA':
+        execution_provider_list = ['cuda']
+    elif execution_provider == 'CPU':
+        execution_provider_list = ['cpu']
+    else:
+        execution_provider_list = ['cpu']
+    print(execution_provider_list)
+
+    # setting paths
+    src_saving_path_complete = os.path.join(saving_path, src_file.filename)
+    target_saving_path_complete = os.path.join(saving_path, target_file.filename)
+    output_saving_path_complete = os.path.join(saving_path, 'output_' + target_file.filename)
+    with open(src_saving_path_complete, "wb+") as file_object:
+        file_object.write(src_file.file.read())
+    with open(target_saving_path_complete, 'wb+') as file_obj:
+        file_obj.write(target_file.file.read())
+
+    roop.globals.source_path = src_saving_path_complete
+    roop.globals.target_path = target_saving_path_complete
+    roop.globals.output_path = output_saving_path_complete
+
+    # Since API starts the roop. It is always headless
+    roop.globals.headless = True
+    # Setting other roop parameters
+    roop.globals.frame_processors = roop_parameters.frame_processor
+    roop.globals.keep_fps = roop_parameters.keep_fps
+    roop.globals.keep_frames = roop_parameters.keep_frames
+    roop.globals.skip_audio = roop_parameters.skip_audio
+    roop.globals.many_faces = roop_parameters.many_faces
+    roop.globals.reference_face_position = roop_parameters.reference_face_position
+    roop.globals.reference_frame_number = roop_parameters.reference_frame_number
+    roop.globals.similar_face_distance = roop_parameters.similar_face_distance
+    roop.globals.temp_frame_format = roop_parameters.temp_frame_format
+    roop.globals.temp_frame_quality = roop_parameters.temp_frame_quality
+    roop.globals.output_video_encoder = roop_parameters.output_video_encoder
+    roop.globals.output_video_quality = roop_parameters.output_video_quality
+    roop.globals.max_memory = roop_parameters.max_memory
```

Comment:
not everything here should be exposed... the memory limit could be an ENV you set via docker-compose.yml